Unable to run 'make test' on osx with BUILD_IN_DOCKER=true

When running 'make test' on osx with BUILD_IN_DOCKER=true gives the following error:

```
$ make BUILD_IN_DOCKER=true test
Building for GOARCH=amd64
Everything is gofmt approved!
go generate -v pkg/controllers/proxy/network_services_controller.go
pkg/controllers/proxy/network_services_controller.go
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:27:2: could not import github.com/docker/libnetwork/ipvs (invalid package name: "")
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:106:112: RT_SCOPE_LINK not declared by package syscall
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:127:112: RT_SCOPE_LINK not declared by package syscall
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:155:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method GetServices
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:159:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method GetDestinations
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:163:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method DelDestination
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:167:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method NewDestination
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:171:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method UpdateDestination
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:175:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method DelService
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:179:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method UpdateService
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:183:9: invalid operation: ln.ipvsHandle (variable of type *invalid type) has no field or method NewService
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1824:16: invalid operation: svc (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1824:44: invalid operation: svc (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1824:68: invalid operation: svc (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1825:23: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1825:67: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1849:20: invalid operation: svc (variable of type *invalid type) has no field or method SchedName
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1850:5: invalid operation: svc (variable of type *invalid type) has no field or method SchedName
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1915:16: invalid operation: svc (variable of type *invalid type) has no field or method FWMark
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1916:23: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1916:67: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1940:20: invalid operation: svc (variable of type *invalid type) has no field or method SchedName
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1941:5: invalid operation: svc (variable of type *invalid type) has no field or method SchedName
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:597:6: invalid operation: ipvsService (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:598:14: invalid operation: ipvsService (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:599:7: invalid operation: ipvsService (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:604:15: invalid operation: ipvsService (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:605:13: invalid operation: ipvsService (variable of type *invalid type) has no field or method FWMark
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:606:56: invalid operation: ipvsService (variable of type *invalid type) has no field or method FWMark
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:666:25: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:668:20: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:668:60: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:675:20: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:675:60: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:688:125: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:689:126: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:690:127: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:691:128: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:692:123: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:693:129: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:694:130: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:695:125: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:696:126: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:697:129: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Stats
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1071:51: FAMILY_V4 not declared by package netlink
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1097:6: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1103:6: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1104:27: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1104:80: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1105:13: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method FWMark
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1106:21: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method FWMark
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1129:15: invalid operation: dst (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1144:9: invalid operation: ipvsSvc (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1145:65: invalid operation: dst (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1145:128: invalid operation: dst (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1148:72: invalid operation: dst (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1148:118: invalid operation: dst (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1151:66: invalid operation: dst (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1151:112: invalid operation: dst (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1723:9: invalid operation: s (variable of type *invalid type) has no field or method Protocol
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1732:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1736:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1740:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1744:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1748:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1752:5: invalid operation: s (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1756:55: invalid operation: s (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1756:66: invalid operation: s (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1760:43: invalid operation: d (variable of type *invalid type) has no field or method Address
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1760:54: invalid operation: d (variable of type *invalid type) has no field or method Port
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1760:62: invalid operation: d (variable of type *invalid type) has no field or method Weight
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1765:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1766:3: invalid operation: svc (variable of type *invalid type) has no field or method Netmask
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1768:3: invalid operation: svc (variable of type *invalid type) has no field or method Timeout
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1770:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1771:3: invalid operation: svc (variable of type *invalid type) has no field or method Netmask
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1772:3: invalid operation: svc (variable of type *invalid type) has no field or method Timeout
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1778:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1780:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1784:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1786:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1790:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1792:3: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1796:6: invalid operation: svc (variable of type *invalid type) has no field or method Netmask
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1797:3: invalid operation: svc (variable of type *invalid type) has no field or method Netmask
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1799:3: invalid operation: svc (variable of type *invalid type) has no field or method Netmask
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1805:18: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1805:59: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1809:18: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1809:59: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1813:18: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:1813:59: invalid operation: svc (variable of type *invalid type) has no field or method Flags
/Users/lucas.mundim/Projects/golang/src/github.com/cloudnativelabs/kube-router/pkg/controllers/proxy/network_services_controller.go:2211:44: FAMILY_V4 not declared by package netlink
couldn't load packages due to errors: github.com/cloudnativelabs/kube-router/pkg/controllers/proxy
moq [flags] destination interface [interface2 [interface3 [...]]]
  -out string
    	output file (default stdout)
  -pkg string
    	package name (default will infer)
pkg/controllers/proxy/network_services_controller.go:95: running "moq": exit status 1
make: *** [pkg/controllers/proxy/network_services_controller_moq.go] Error 1
```

The import github.com/docker/libnetwork/ipvs from pkg/controllers/proxy/network_services_controller.go only builds on linux so 'go generate' from %_moq.go target needs to run inside docker when BUILD_IN_DOCKER is true.

--- a/Makefile
+++ b/Makefile
@@ -151,8 +151,13 @@ gomoqs: ./pkg/controllers/proxy/network_services_controller_moq.go
 # file_moq.go file is generated from file.go "//go:generate moq ..." in-file
 # annotation, as it needs to know which interfaces to create mock stubs for
 %_moq.go: %.go
+ifeq "$(BUILD_IN_DOCKER)" "true"
+	$(DOCKER) run -v $(PWD):/go/src/github.com/cloudnativelabs/kube-router -w /go/src/github.com/cloudnativelabs/kube-router $(DOCKER_BUILD_IMAGE) \
+			sh -c 'apk add --no-cache git build-base && go get github.com/matryer/moq && go generate -v $(*).go'
+else
 	@test -x $(GOPATH)/bin/moq && exit 0; echo "ERROR: 'moq' tool is needed to update mock test files, install it with: \ngo get github.com/matryer/moq\n"; exit 1
 	go generate -v $(*).go
+endif
 
 gopath: ## Warns about issues building from a directory that does not match upstream.
 	@echo 'Checking project path for import issues...'

